### PR TITLE
add block and timeout arguments of  function

### PR DESCRIPTION
**Pull request by zenghsh3**

### CODE REVIEW

**Comment by zenghsh3** [comment]
fixed #542

File: parl/remote/future_mode/future_object.py [hunk 1]
```diff
@@ -53,14 +54,27 @@ def func(self):
         self._already_get = False
         self.internal_lock = threading.Lock()
 
-    def get(self):
+    def get(self, block=True, timeout=None):
```
Comment (TomorrowIsAnOtherDay) [inline parl/remote/future_mode/future_object.py:57]: Great! This api is consistent with the queue APIs in Python.

File: parl/remote/exceptions.py [hunk 2]
```diff
@@ -97,3 +97,15 @@ def __init__(self, func_name):
...
         return self.error_info
+
+
+class FutureObjectEmpty(Exception):
+    """
+    Exception raised when non-blocking get() (or get_nowait()) is called on a `FutureObject` which is empty.
+    """
+
+    def __init__(self):
+        self.error_info = "The `FutureObject` is empty."
```
Comment (TomorrowIsAnOtherDay) [inline parl/remote/exceptions.py]: Please update error info.

**Review by TomorrowIsAnOtherDay — APPROVED** [review]
Looks good to me.

**Pull request merged**